src/main/java/de/hpi/datastreams/apps/BaseKafkaApp.java

```diff
@@ -48,6 +49,12 @@ private Properties getProperties() {
         props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, JSONSerde.class.getName());
         props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
 
+        props.put(StreamsConfig.PRODUCER_PREFIX + ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
+                "io.confluent.monitoring.clients.interceptor.MonitoringProducerInterceptor");
+        props.put(
+                StreamsConfig.CONSUMER_PREFIX + ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG,
+                "io.confluent.monitoring.clients.interceptor.MonitoringConsumerInterceptor");
+
```

Comment:
I am not really understanding what this does. Maybe you can explain it to me.

Reply:
This is needed to be able to track the actual message flow within Confluent's ControlCenter